integration-tests/src/test/groovy/com/okta/sdk/tests/it/GroupRolesIT.groovy

```diff
@@ -0,0 +1,195 @@
+        Role userAdminRole = createdGroup.assignRole(userAdminRoleRequest)
+        Role appAdminRole = createdGroup.assignRole(appAdminRoleRequest)
+
+        // 5. List assigned group roles and check the assignments
+        RoleList roles = client.listGroupAssignedRoles(createdGroup.getId())
+
+        assertRolePresent(roles, userAdminRole)
+        assertRolePresent(roles, appAdminRole)
+    }
+
+    @Test
+    @Scenario("unassigned-roles-for-group")
+    void unassignRoleForGroupTest() {
+
+        String groupName = "sdk-java List Test Group ${uniqueTestName}"
+
+        // 1. Create a new group
+        Group createdGroup = GroupBuilder.instance()
+            .setName(groupName)
+            .buildAndCreate(client)
+        registerForCleanup(createdGroup)
+
+        validateGroup(createdGroup, groupName)
+
+        // 2. List all groups and find the group created
+        assertGroupPresent(client.listGroups(), createdGroup)
+
+        // 3. Create a USER_ADMIN role request
+        AssignRoleRequest userAdminRoleRequest = client.instantiate(AssignRoleRequest)
+        userAdminRoleRequest.setType(RoleType.USER_ADMIN)
+
+        // 4. Assign the above role
+        Role userAdminRole = createdGroup.assignRole(userAdminRoleRequest)
+
+        // 5. List assigned group roles and check the assignments
+        assertRolePresent(client.listGroupAssignedRoles(createdGroup.getId()), userAdminRole)
+
+        // 6. Remove the role from group
+        client.removeRoleFromGroup(createdGroup.getId(), userAdminRole.getId())
+
+        // 7. List assigned group roles and check the assignments
+        assertRoleAbsent(client.listGroupAssignedRoles(createdGroup.getId()), userAdminRole)
+    }
+
+    @Test
+    @Scenario("list-group-targets-for-group")
+    void listGroupTargetsForGroupTest() {
+
+        String groupName1 = "sdk-java List Test Group ${uniqueTestName}-1"
+        String groupName2 = "sdk-java List Test Group ${uniqueTestName}-2"
+
+        // 1. Create two groups
+        Group createdGroup1 = GroupBuilder.instance()
+            .setName(groupName1)
+            .buildAndCreate(client)
+        Group createdGroup2 = GroupBuilder.instance()
+            .setName(groupName2)
+            .buildAndCreate(client)
+        registerForCleanup(createdGroup1)
+        registerForCleanup(createdGroup2)
+
+        validateGroup(createdGroup1, groupName1)
+        validateGroup(createdGroup2, groupName2)
+
+        // 2. List all groups and find the groups created
+        assertGroupPresent(client.listGroups(), createdGroup1)
+        assertGroupPresent(client.listGroups(), createdGroup2)
+
+        // 3. Create a USER_ADMIN role & assign it to the first group
+        Role userAdminRole = createdGroup1.assignRole(client.instantiate(AssignRoleRequest)
+            .setType(RoleType.USER_ADMIN))
+
+        // 4. Add second group as the admin target for the user admin role
+        userAdminRole.addAdminGroup(createdGroup2.getId())
```

Comment:
My understanding is that [Add a group target to a group admin role endpoint](https://developer.okta.com/docs/reference/api/roles/#add-a-group-target-to-a-group-administrator-role) needs three inputs:
```
/api/v1/groups/${groupId}/roles/${roleId}/targets/groups/${targetGroupId}
```
where do you get group1 reference?

Reply:
Good question. 

First, I create a group (createdGroup1) and then invoke `assignRole` operation on the created group instance.

`Role userAdminRole = createdGroup1.assignRole(client.instantiate(AssignRoleRequest)
            .setType(RoleType.USER_ADMIN))`

Next, I associate the group target (createdGroup2) to the above created admin role:

`userAdminRole.addAdminGroup(createdGroup2.getId())`

Hope this clarifies.

Reply:
> > Good question.
> > First, I create a group (createdGroup1) and then invoke `assignRole` operation on the created group instance.
> > `Role userAdminRole = createdGroup1.assignRole(client.instantiate(AssignRoleRequest) .setType(RoleType.USER_ADMIN))`
> 
> This is clear ^.
> 
> > Next, I associate the group target (createdGroup2) to the above created admin role:
> > `userAdminRole.addAdminGroup(createdGroup2.getId())`
> 
> This is what I am missing^. How do you translate these 2 params to the next href?
> 
> ```
> /api/v1/groups/${groupId}/roles/${roleId}/targets/groups/${targetGroupId}
> ```
> 
> I can imagine you are doing this:
> 
> ```
> /api/v1/groups/${groupId}/roles/${userAdminRole.GetId()}/targets/groups/${createdGroup2.getId()}
> ```
> 
> but how do you map group 1 here? what about if you have previously assigned more than one group to the same role? How do you make sure you are adding a group target to the corresponding group?
> 

Okta core returns the `groupId` in the `href` of the response for `assignRole` operation. 
I use the groupId from there.

> In .NET we have something like this:
> 
> ```
> client.Groups.AddGroupTargetToGroupAdministratorRoleForGroupAsync(createdGroup1.Id, role.Id, createdGroup2.Id);
> ```
> 
> makes sense?
> 
> > Hope this clarifies.

Reply:
After discussing this offline, it makes sense now. My only suggestion is to rename `addAdminGroup` to something like `addAdminGroupTarget`, so it is clear you are adding a group target.

Reply:
Updated.